### PR TITLE
Column names test

### PR DESCRIPTION
**Pull request by FinchPowers**
Ready for review.

### CODE REVIEW

File: testing/column_name_test.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    def test_float_arith(self):
+        self.select('1 + 1.1', '"1 + 1.1"')
+
+    def test_col_arith(self):
+        self.select('a + b FROM (SELECT a:1, b:2)', 'a + b')
+
+    @unittest.expectedFailure
+    def test_obj_arith(self):
+        self.select('a:{x:1} + 1', 'a:{x:1} + 1.x')  # currently returns a.x
```
Comment (FinchPowers) [inline testing/column_name_test.py]: Pretty sure this can be considered a bug.

Comment (guyd) [inline testing/column_name_test.py]: And what is the value return in this case?

File: testing/column_name_test.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        self.select("{x:1}", '{x:1}.x')
+
+    def test_object_as_star(self):
+        self.select("{x:1} AS *", 'x')
+
+    def test_object_as_x(self):
+        self.select("{x:1} AS x", 'x.x')
+
+    def test_object_over_object_arith(self):
+        self.select("{x:1} + {x:1}", '{x:1} + {x:1}.x')  # <--- ugly
```
Comment (guyd) [inline testing/column_name_test.py]: ugly but consistent

File: testing/column_name_test.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    def test_object_extraction(self):
+        self.select("col.x FROM (SELECT col.x:1)", 'col.x')
+        self.select("* FROM (SELECT col.x:1)", 'col.x')
+        self.select("col.* FROM (SELECT col.x:1)", 'col.x')
+
+    def test_object_extrac_star(self):
+        self.select("c* FROM (SELECT col.x:1)", 'col.x')
+        self.select("c* AS * FROM (SELECT col.x:1)", 'x')
```
Comment (guyd) [inline testing/column_name_test.py]: Interesting! I did not know that this was allowed.

File: testing/column_name_test.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        self.select("* AS * FROM (SELECT col.x:1)", 'col.x')
+
+    def test_extract_all(self):
+        self.select("{*} FROM (SELECT col.x:1)", '{*}.col.x')
+        self.select("{*} AS * FROM (SELECT col.x:1)", 'col.x')
+
+#     def test_object_arith(self):
```
Comment (guyd) [inline testing/column_name_test.py]: May be this deserve a separate test where you would also check the return value and not just the column name.

**Comment by guyd** [comment]
Such a test is definitively useful.  I agree that the highlighted line looks like a bug.

File: testing/column_name_test.py (hunk 1 above)
Comment (FinchPowers) [inline testing/column_name_test.py]: C'est écrit en commentaire à côté! :)
`a.x`

**Comment by FinchPowers** [comment]
Il y a ce test aussi qui est complémentaire. Il opère sur un dataset.
https://github.com/mldbai/mldb/blob/master/testing/MLDB-832-select_star.py

File: testing/column_name_test.py (hunk 1 above)
Comment (guyd) [inline testing/column_name_test.py]: Not the column name, the value.

Comment (FinchPowers) [inline testing/column_name_test.py]: Ah! Il fait + 1 aux feuilles.

File: testing/testing.mk [hunk 5]
```diff
@@ -419,6 +419,8 @@ $(eval $(call mldb_unit_test,MLDB-1755-column-execution-memory-use.js))
 $(eval $(call mldb_unit_test,stddev_builtin_fct_test.py))
 $(eval $(call mldb_unit_test,MLDB-1810-new-executor-rowpath.js))
 $(eval $(call mldb_unit_test,MLDB-1827_operator_null_propagation_test.py))
+$(eval $(call mldb_unit_test,column_name_test.py))
+$(eval $(call mldb_unit_test,MLDB-1834_select_row_expr_star_err_msg.py))
```
Comment (guyd) [inline testing/testing.mk]: Duplicated line here.

**Comment by guyd** [comment]
+1 once the duplicated line is removed.

File: testing/testing.mk (hunk 5 above)
Comment (FinchPowers) [inline testing/testing.mk]: Bien vu!

**Pull request merged**